pom.xml

```diff
@@ -19,28 +19,158 @@
   <properties>
     <github.repository>IDEasy</github.repository>
     <ide_version>${revision}</ide_version>
-    <java.version>17</java.version>
+    <java.version>21</java.version>
+    <maven.compiler.source>21</maven.compiler.source>
+    <maven.compiler.target>21</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <jline.version>3.24.1</jline.version>
+    <jansi.version>2.4.1</jansi.version>
+    <jackson.version>2.18.2</jackson.version>
+    <xmlunit.version>2.10.0</xmlunit.version>
+    <wiremock.version>3.11.0</wiremock.version>
+    <javafx.version>21</javafx.version>
+    <testfx.version>4.0.18</testfx.version>
   </properties>
 
-  <dependencies>
-    <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
-      <version>5.12.0</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.assertj</groupId>
-      <artifactId>assertj-core</artifactId>
-      <version>3.24.2</version>
-      <scope>test</scope>
-    </dependency>
-  </dependencies>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-compress</artifactId>
+        <version>1.27.1</version>
+      </dependency>
+      <dependency>
+        <groupId>jakarta.json</groupId>
+        <artifactId>jakarta.json-api</artifactId>
+        <version>2.1.3</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.parsson</groupId>
+        <artifactId>parsson</artifactId>
+        <version>1.1.7</version>
+      </dependency>
+      <dependency>
+        <groupId>com.fasterxml.jackson.core</groupId>
+        <artifactId>jackson-databind</artifactId>
+        <version>${jackson.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>com.fasterxml.jackson.datatype</groupId>
+        <artifactId>jackson-datatype-jsr310</artifactId>
+        <version>${jackson.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>com.fasterxml.jackson.dataformat</groupId>
+        <artifactId>jackson-dataformat-xml</artifactId>
+        <version>${jackson.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter</artifactId>
+        <version>5.12.0</version>
+        <scope>test</scope>
```

Comment:
Can you remove the `scope` (and also out-commented `optional`) from `dependencyManagement`?
This is inherited into dependencies then.
While several projects do that for test dependencies to omit the `<scope>test</scope>` in the real `<dependency>`, this can cause a lot of confusion.
The maven default scope if omitted is `compile`, so if the scope is missing, then I assume it is compile.
Even if you would consistently do that for all test dependencies the problem is that sometimes you need to create a module for shared and reusable test-code and there you need `junit`, etc. as compile-dependency.

Anyhow, you always included `<scope>test</scope>` in the real `<dependency>` elements in the child POMs.
Therefore I would recommend to remove them always here.

Reply:
Replaced with constructive review comment.